var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -83,30 +83,16 @@ class Proj(AutotoolsPackage):
     )
 
     # https://proj.org/install.html#build-requirements
-    depends_on("pkgconfig@0.9.0:", type="build", when="@6:")
```

Comment:
Need to specify required versions of cmake

Reply:
I've made the change w.r.t. to the `cmake_minimum_version` for the project but I'm curious, is this a standard convention for CMake packages or is this a more specific requirement to proj? I just ask because I've ported/am porting a number of CMake projects and have yet to notice a consideration for minimum version/consider it myself and I'll need to go and correct that.

Reply:
All CMake projects do this and should list the correct dependency range in `package.py`: https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#important-files

Reply:
Is this something we should be enforcing then? A quick grep of the packages directory reveals a not insignificant number of CMake packages that do not do this.

Reply:
The majority of people who review PRs and merge them don't know what they're doing lol. This kind of CMake version thing isn't the biggest issue (Spack defaults to the latest version anyway), but if you have an old version installed (maybe as an external package) then Spack will happily use it unless the correct dependency requirements are listed. Basically, there's no reason not to be more specific about the dependency, and it can prevent issues for users too.

Reply:
I completely agree, it's a very good idea, and I definitely fall into the category of people who this helps (I think I have cmake like 2.8 somewhere on my machine for some test I did once) and I'm always a fan of being more explicit whenever possible, so I have no problem with the policy/suggestion at all, I was more just wondering if this was something we could/should add to the package audits check for cmake packages? Seems worth enforcing IMO as the minimum version is pretty important, and it would take the burden off those reviewers who aren't aware of the suggestion/policy.

Reply:
Not sure how easy this would be to add to the audit tests. Not all packages declare a minimum version in CMakeLists.txt so we would have to download and grep the source code.

Reply:
True, it's not mandatory, but I think all the audit test would need to report is that the package recipe does not specify a CMake version constraint for the package, and at that point it doesn't seem unreasonable to ask someone creating the package recipe to know this fairly basic piece of info about the project they're adding to spack/converting to CMake or to find out. Maybe I'm just not familiar enough with other use cases for Spack, but it seems like if you're adding/modifying a package, you're at least able to access that package's CMakeLists.txt, or if you're not familiar with CMake, set an arbitrary very low version. That last scenario isn't really different from the current setup, but at least kind of forces the consideration (which is all that's really important IMO).

Reply:
This is also something I'd be happy to poke at myself, not trying to suggest more work for someone else lol.

Reply:
Not sure what that arbitrary low version would be. I guess I'm not sure how common it is for developers to set this flag.

Reply:
I think just setting something like `cmake@2` would probably be sufficiently low. I personally have never seen a project fail to set the minimum CMake version, although I'm absolutely sure some projects do. It's generally considered best practice to include it, that being said, assuming best practices are ever followed is not realistic, but in this case I think it's common enough.

Reply:
I think newer versions of CMake started warning when a minimum isn't set, so it's def more common that it used to be.